File: 4038-count-integers-appearing-in-a-single-block/4038-count-integers-appearing-in-a-single-block.java
class Solution {
    public int countSpecialIntegers(int[] nums) {
        HashSet<Integer> set = new HashSet<>();
        HashSet<Integer> check = new HashSet<>();

        int count = 0;
        int i = 0;

        while (i < nums.length) {
            int j = i + 1;
            while (j < nums.length && nums[j] == nums[i]) j++;

            if (!set.contains(nums[i])) {
                set.add(nums[i]);
                count++;
            } else if (!check.contains(nums[i])) {
                check.add(nums[i]);
                count--;
            }

            i = j;
        }

        return count;
    }
}

File: 4038-count-integers-appearing-in-a-single-block/4038-count-integers-appearing-in-a-single-block.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var countSpecialIntegers = function(nums) {
    const set = new Set();
    const check = new Set();

    let count = 0;
    let i = 0;

    while (i < nums.length) {
        let j = i + 1;
        while (j < nums.length && nums[j] == nums[i]) j++;

        if (!set.has(nums[i])) {
            set.add(nums[i]);
            count++;
        } else if (!check.has(nums[i])) {
            check.add(nums[i]);
            count--;
        }
        i = j;
    }
    return count;
};